Project/DAQ_System/Core/Src/main.c

```diff
@@ -96,7 +100,29 @@ int main(void)
   MX_USART3_UART_Init();
   MX_USB_OTG_FS_PCD_Init();
   MX_ADC1_Init();
+  MX_CAN1_Init();
   /* USER CODE BEGIN 2 */
+
+  	// CAN initialization
+	HAL_CAN_Start(&hcan1); // starts the CAN bus
+	HAL_CAN_ActivateNotification( &hcan1, CAN_IT_RX_FIFO0_MSG_PENDING); // activate interrupt notifications
+
+
+	CAN_FilterTypeDef CAN;
+	CAN.FilterIdHigh = 0x103;
+	CAN.FilterIdLow = 0;
+	CAN.FilterMaskIdHigh = 0xFFFF;
+	CAN.FilterMaskIdLow = 0;
+	CAN.FilterFIFOAssignment = CAN_FILTER_FIFO0;
+	CAN.FilterBank = 13;
+	CAN.FilterMode = CAN_FILTERMODE_IDMASK;
+	CAN.FilterScale = CAN_FILTERSCALE_16BIT;
+	CAN.FilterActivation = CAN_FILTER_ENABLE;
+
+	HAL_CAN_ConfigFilter(&hcan1,&CAN);
```

Comment:
I would prefer that you moved code that is not auto-generated by CubeMX to the `app.cpp`.

Reply:
moved it inside cppMain on app.cpp